lib/profileAgents.js

```diff
@@ -602,6 +608,105 @@ async function _updateMeterController({meterId, controller}) {
   }
 }
 
+async function _getProfileAgentRecords({accountId, includeSecrets}) {
+  const query = {'profileAgent.account': accountId};
+  const projection = {_id: 0};
+  const collection = getCollection('profile-profileAgent');
+  let records = await collection.find(query, {projection}).toArray();
+  const zcapRefreshed = await _refreshProfileAgentZcaps({
+    profileAgentRecords: records
+  });
+  if(zcapRefreshed) {
+    records = await collection.find(query, {projection}).toArray();
+  }
+  if(!includeSecrets) {
+    records = await removeSecretsFromRecords({records});
+  }
+  return records;
+}
+
+async function _refreshProfileAgentZcaps({profileAgentRecords}) {
+  let zcapRefreshed = false;
+  const records = JSON.parse(JSON.stringify(profileAgentRecords));
```

Comment:
I believe `klona` is the preferred copy method.  It's already a dependency in this package.

Reply:
Thanks, updated to use klona here [925e31f](https://github.com/digitalbazaar/bedrock-profile/pull/99/commits/925e31f8c9cdfc0b268125829b6f7d6a141ac608)